src/cpp/rtps/builtin/liveliness/WLPListener.cpp

```diff
@@ -74,22 +74,28 @@ void WLPListener::onNewCacheChangeAdded(
             break;
         }
     }
-    if (change->serializedPayload.length > 0)
+
+    // Data should have at least 4 bytes of representation header, 12 of GuidPrefix, and 4 of kind.
+    if (change->serializedPayload.length >= 20)
     {
-        if (PL_CDR_BE == change->serializedPayload.data[1])
+        // Encapsulation in the second byte of the representation header.
+        change->serializedPayload.encapsulation = (uint16_t)change->serializedPayload.data[1];
+
+        // Extract GuidPrefix
+        memcpy(guidP.value, change->serializedPayload.data + 4, 12);
+
+        // Extract liveliness kind
+        if (is_wlp_kind(&change->serializedPayload.data[16]))
```

Comment:
Would be better if this method has an output argument that returns the liveliness kind.
We would then only have the failing validation condition:

```
if (!get_wlp_kind(&change->serializedPayload.data[16], livelinessKind))
{
    // logInfo + remove_change + return
}
```